_posts/getting-started/2024-03-04-using-the-cli-linux.md

```diff
@@ -0,0 +1,177 @@
+[Using the OpenStack CLI (Windows)]({{ '/articles/using-the-cli-windows' | relative_url }}).
+
+The OpenStack Command Line Interface (CLI) is a powerful tool for managing
+OpenStack resources. This guide will show you how to install the OpenStack CLI,
+log in to the OpenStack CLI, and use the OpenStack CLI to manage OpenStack
+resources.
+
+This guide makes use of the `clouds.yaml` file to store OpenStack credentials.
+You can also use environment variables to store your credentials using the
+`openrc` file. The `clouds.yaml` file is the recommended way to store your
+OpenStack credentials, as it is easier to manage and add multiple clouds
+(projects, regions etc.).
+
+---
+
+## Install the OpenStack CLI
+The installation of the OpenStack CLI is different for each Operating System 
+and distrubution. Below you will find the installation instructions for the
+most common Linux distributions.
+
+For Windows installations please see
+[Using the OpenStack CLI (Windows)]({{ '/articles/using-the-cli-windows' | relative_url }}).
+
+
+### Debian (10, 11, 12) | Ubuntu (20.04 | 22.04)
+To install the OpenStack CLI, run the following commands:
+
+**Step 1**  
+Update our package list to make sure we install the latest version of the
+OpenStack CLI.
+```bash
+sudo apt update
+```
+
+**Step 2**  
+Install the OpenStack CLI
+```bash
+sudo apt install python3-openstackclient
+```
+
+After the installation has finished proceed to
+[Preparing your OpenStack Credentials](#preparing-your-openstack-credentials).
+
+### CentOS Stream (8, 9) | RHEL (8, 9) | Rocky Linux (8, 9) | AlmaLinux (8, 9)
+
+To install the OpenStack CLI, run the following commands:
+
+**Step 1**  
+Install python3 and pip3 
+```bash
+sudo dnf install python3 python3-pip
+```
+
+**Step 2**  
+We will now update pip3 to make sure we use the latest version of pip3
+```bash
+sudo pip3 install --upgrade pip
+```
+
+**Step 3**  
+Now we can install the OpenStack CLI
+```bash
+sudo pip3 install python-openstackclient
+```
+
+After the installation has finished proceed to
+[Preparing your OpenStack Credentials](#preparing-your-openstack-credentials).
+
+---
+
+### Preparing Your OpenStack Credentials
+After installing the OpenStack CLI, we need to prepare our credentials to make
+sure we are able to login using the OpenStack CLI.
+
+**Step 1**  
+First we will create a file to store our OpenStack credentials. In this
+tutorial we will use the `clouds.yaml` file to store our credentials. You
+can also use environment variables to store your credentials using the
+`openrc` file.
+
+Create the directory in which the clouds.yaml file will be stored.
+```bash
+mkdir -p ~/.config/openstack
+```
+
+**Step 2**  
+We will now create the clouds.yaml file, the next step will describe how to
+populate the file with the necessary information.
+
+```bash
+nano ~/.config/openstack/clouds.yaml
+```
+
+**Step 3**  
+In the clouds.yaml file add the following content to it. Replace: `region`,
```

Comment:
Horizon should (depending on version) be able to provide a clouds.yml file as quickstart, maybe include this too in the documentation? 

https://opendev.org/openstack/horizon/commit/6d315598aa29ddda8dd3090170f2ef7d030e13ef

Reply:
@mnederlof I am aware of the quickstart download but not all clouds/environments have this enabled just yet thus I chose to use the API Access page instead of the QuickStart menu in the top right corner.

I chose to add the clouds.yaml file as is in the tutorial because when you download the normal clouds.yaml you won't find a password field which I chose to include. Without the password field the OpenStack client keeps asking for you password when executing a command, although more secure also really annoying to use.

Reply:
Maybe just add a small comment about it?